packages/swapper/src/swappers/PortalsSwapper/utils/fetchPortalsTradeOrder.ts

```diff
@@ -81,12 +82,15 @@ export const fetchPortalsTradeOrder = async ({
   feePercentage,
   validate,
   swapperConfig,
+  hasWallet,
 }: PortalsTradeOrderParams): Promise<PortalsTradeOrderResponse> => {
   const url = `${swapperConfig.REACT_APP_PORTALS_BASE_URL}/v2/portal`
 
+  if (hasWallet && !sender) throw new Error('Missing sender address')
+
   const params = new URLSearchParams({
     partner,
-    sender,
+    ...(sender ? { sender } : {}),
```

Comment:
remark: `{ [key]: undefined }` not being valid here is why i hate (type|java)script

Reply:
srsly